### ISSUE TITLE
package regressions in 7.8

### ISSUE THREAD

**Issue by samth** [racket/racket#3300]
Currently, the following packages either fail to build, or have test failures, regressing from 7.7. 

7.7 build: https://pkg-build.racket-lang.org
7.8 pre-release build: https://plt.eecs.northwestern.edu/release-pkg-build/

[EDIT] using check-boxes to indicate "IMO not a release blocker (anymore)"

### Genuine regressions

- [x] clang, digimon, graphics, lambda-sh, psd, pydrnlp, ricoeur-tei-utils, schema, w3s
    * Failure is here: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/fail/digimon.txt
    * Problem is likely related to one of the struct-info changes (cc @sorawee)
- [x] planning
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/fail/planning.txt
    * xexpr error caused by a bug in racket/scribble#232. (cc @jackfirth)
- [x] doc-coverage
    * Failure is here: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/test-fail/doc-coverage.txt
    * Missing documentation for `for-clause-syntax-protect` `syntax-pattern-variable?`. Are these supposed to be exported?
- [x] lwc2016
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/fail/lwc2016.txt
    * New `translate` identifier provided by `slideshow` causes a conflict (introduced in racket/pict#57 by @florence)
- [x] measures-with-dimensions
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/fail/measures-with-dimensions.txt
    * Problem with submodule path access in Typed Racket. Possible that racket/typed-racket#907 is related (cc @bennn)
- [x] timable
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/test-fail/timable.txt
    * a date test that fails when run between 23:00 and 0:00. (cc @yanyingwang)

### Possible build system bugs

- [x] openweather, tjson
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/fail/openweather.txt
    * Seems to be a permission error inside documentation building

### pkg-build issues
- [x] egg-herbie-linux, egg-herbie-osx, egg-herbie-windows
    * Likely a transient failure (something changed between fetching the sha and fetching the package)
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/archive-fail/egg-herbie-linux.txt
- [x] ipoe
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/test-fail/ipoe.txt
    * Appears to want network access during testing. (cc @bennn)
- [x] racket-weekend
    * Checksum mismatch, probably a transient error
    * [EDIT] wrote to Jesse, he says he updated the zip but didn't update the checksum. He's planning to
        set the package up to use a git URL instead, solving this problem. 
- [x] rwind
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/test-fail/rwind.txt
    * Probably due to different system packages on the two machines (maybe use docker-encapsulated builds to fix?)
- [x] sass
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/test-fail/sass.txt
    * Fails to find foreign library, but I think that should be installed by one of its dependencies. Likely an issue with pkg-build rather than with with package itself.
- [x] amap
    * Failure: https://plt.eecs.northwestern.edu/release-pkg-build/server/built/archive-fail/amap.txt
    * Odd failure to download the package from github.

**Comment by sorawee** [comment]
Some of these are apparently related to `struct-field-info`.

**Comment by jbclements** [comment]
@sorawee Sorry, I should know what happened here: raco docs suggests that struct-field-info-list is new in this release. Am I right in thinking that you added it? If so, do you have a minute (okay, ten minutes (okay, more than ten minutes)) to take a look at one of the failing packages to see what's going on?

**Comment by sorawee** [comment]
Yes, I talked to Sam about this and will take a look at it this afternoon.

**Comment by jbclements** [comment]
Awesome! It looks like maybe the expansion of define-struct fails for certain kinds of uses. Many thanks.

**Comment by samth** [comment]
`ipoe` has a bunch of different test failures on my machine (including postgres password trouble) but those particular failures go away.

**Comment by samth** [comment]
The `rwind` tests all pass on the release candidate on my machine.

**Comment by samth** [comment]
`egg-herbie-linux` installs and passes test on my machine.

**Comment by samth** [comment]
`sass` installs and passes tests on my machine.

**Comment by samth** [comment]
The issue with `tjson` replicates for me on the pre-release.

**Comment by samth** [comment]
The failure with `doc-coverage` is odd. Those identifiers have been exported for quite a while (and are undocumented) so something else must have changed.

**Comment by AlexKnauth** [comment]
The `measures-with-dimensions` regression should be fixed now, by https://github.com/AlexKnauth/measures-with-dimensions/commit/fc6c78f79ac89cf488a5ccc5fc20391bd254886c

**Comment by samth** [comment]
I'm glad it's working for you now, but was the old or new behavior the right one?

**Comment by AlexKnauth** [comment]
The new behavior is right for the `module` in `(require/untyped-contract module [name subtype] ...)`, in my opinion. It is only as wrong as it was before for the `require-form` in `(require/untyped-contract (begin require-form) module [name subtype] ...)`, and I don't know whether it's possible to fix that.

**Comment by sorawee** [comment]
@samth, @jbclements "clang, digimon, graphics, lambda-sh, psd, pydrnlp, ricoeur-tei-utils, schema, w3s" won't be blocked once #3301 is merged into the release branch.

**Comment by rfindler** [comment]
Pushed https://github.com/dfeltey/lwc2016/commit/8c0a6e11f14af23dcbd72890a51d4fd77350a3d7 to avoid the problem with the new export from pict.

**Comment by bennn** [comment]
I disabled more tests in `ipoe` when `CI=true`, so that package shouldn't ask for network access anymore.

**Comment by samth** [comment]
I've opened a PR that should fix the `planning` issue.

**Comment by sorawee** [comment]
I created a PR that should fix the `doc-coverage` problem. The real problem with `doc-coverage` is that it expects `expand-for-clause` to be one of the "uncovered" case, but this is no longer the case due to #3246.

**Comment by sorawee** [comment]
`doc-coverage` should be fixed now.

**Comment by samth** [comment]
The most recent release pkg-build has these "worse" results:
```
  #hash(("doc-coverage" . (success test-fail))
        ("ipoe" . (needs-extra-deps test-fail))
        ("openweather" . (success build-fail))
        ("planning" . (success build-fail))
        ("racket-weekend" . (test-fail build-fail))
        ("rwind" . (no-docs test-fail))
        ("sass" . (success test-fail))
        ("tjson" . (test-fail build-fail)))
```

I expect `doc-coverage`, `ipoe`, `planning` to be fixed in the next build. `rwind` and `sass` are issues with the builder itself, `racket-weekend` still needs a checksum update. 

`tjson` and `openweather` still seem to be some sort of documentation build bug.

**Comment by mflatt** [comment]
Here's a smaller version of the `tjson` doc problem:

```
#lang racket/base
(require racket/sandbox)
(make-evaluator 'typed/racket)
```

Do you know offhand whether something about `typed/racket` or a dependency changed so that it now depends on `string-constants`?

**Comment by mflatt** [comment]
I get the same result for v7.7 and git head with
```
#lang racket/base
(require syntax/moddep)

(show-import-tree 'typed-racket/core
                  #:dag? #t
                  #:path-to 'string-constants/string-constant)
```
where the output is
```
/Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/core.rkt 0
 /Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/private/with-types.rkt 0
  /Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/typecheck/tc-toplevel.rkt 0
   /Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/typecheck/tc-structs.rkt 0
    /Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/typecheck/typechecker.rkt 0
     /Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/typecheck/check-unit-unit.rkt 0
      /Users/mflatt/plt/extra-pkgs/typed-racket/typed-racket-lib/typed-racket/base-env/base-special-env.rkt 0
       /Users/mflatt/plt/extra-pkgs/string-constants/string-constants-lib/string-constants/string-constant.rkt 0
```
So, maybe it's not a simple dependency question.

**Comment by mflatt** [comment]
Eliminating `typed/racket` from the equation, this one errors in head and not v7.7:
```
#lang racket/base
(require racket/sandbox)
(make-evaluator 'racket/base #:requires '(string-constants/string-constant))
```

**Comment by mflatt** [comment]
I see that `string-constants/string-constant` changed to use `setup/getinfo` instead of direct `requires` to a few languages. It would probably be a good idea to build the relevant table on demand instead of at module instantiation time, but I don't know if that change would be enough. @rfindler

**Comment by rfindler** [comment]
Oh, okay. I'll have to think about this. This was the change to make the LGPL string constants move out into another package, with only a dynamic dependency. Apparently not dynamic enough. I think I can make it always load the built-in ones and then load any other ones later.

I'm not really clear on the requirements here, however. I'll try to make this work at least:

```
#lang racket/base
(require racket/sandbox)
(make-evaluator 'typed/racket)
```

**Comment by rfindler** [comment]
Okay, I've pushed [a commit](https://github.com/racket/string-constants/commit/0b1de4ce3c4ee52db5e2556982a8fbcf8f22ee12) to string-constants that seems to improve `tjson`. I see these messages during the docs build, however:

```
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:32:10: WARNING: no declared exporting libraries for definition
  in: Json
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:37:10: WARNING: no declared exporting libraries for definition
  in: JsObject
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:41:10: WARNING: no declared exporting libraries for definition
  in: JsList
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:45:10: WARNING: no declared exporting libraries for definition
  in: JsNull
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:54:10: WARNING: no declared exporting libraries for definition
  in: string->json
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:67:10: WARNING: no declared exporting libraries for definition
  in: jsobject
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:81:10: WARNING: no declared exporting libraries for definition
  in: jsattribute?
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:85:10: WARNING: no declared exporting libraries for definition
  in: jsattribute
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:91:10: WARNING: no declared exporting libraries for definition
  in: jsattribute-orelse
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:101:10: WARNING: no declared exporting libraries for definition
  in: jsobject-add!
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:105:10: WARNING: no declared exporting libraries for definition
  in: jsobject-remove!
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:110:10: WARNING: no declared exporting libraries for definition
  in: jsobject-update!
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:122:10: WARNING: no declared exporting libraries for definition
  in: json->string
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:132:10: WARNING: no declared exporting libraries for definition
  in: write-json
/Users/robby/git/exp/plt/racket/share/pkgs/tjson/scribblings/tjson.scrbl:136:10: WARNING: no declared exporting libraries for definition
  in: read-json
```

**Comment by sorawee** [comment]
[7.7](https://pkg-build.racket-lang.org/server/built/install/tjson.txt) also has the same warnings, so that shouldn't be an issue, right? If so, then `tjson` is now resolved, too.

**Comment by rfindler** [comment]
Ah okay.

The build will run tonight and we'll see by tomorrow night I guess.

**Comment by samth** [comment]
The latest snapshot installs tjson and openweather properly.

**Comment by samth** [comment]
`ipoe` continues to have test failures.

**Comment by samth** [comment]
`planning` will be fixed, but the relevant change hasn't been merged to racket/scribble@release yet.

**Comment by bennn** [comment]
I pushed an update to `ipoe`.

That test was inside a handler for `exn:fail:network?` exceptions, though. Is the tcp-connect error not a network exception? (`tcp-connect: connection failed`)

**Comment by samth** [comment]
```
> (define x (with-handlers ([values values]) (tcp-connect "google.coom" 25)))
> x
(exn:fail:network:errno
 "tcp-connect: connection failed;\n host not found\n  address: google.coom\n  port number: 25\n  system error: Name or service not known; gai_err=-2"
 #<continuation-mark-set>
 '(-2 . gai))
> (exn:fail:network? x)
#t
```

**Comment by bennn** [comment]
Could the release be different?

The test failure complains about `db.rkt:1701:8`, and that line was inside a with-handlers.

<https://github.com/bennn/iPoe/blob/a4b4462fb3f890d242fb1ccf281e43fd116b76d5/ipoe/private/db.rkt#L1701>

<details>

```
db.rkt: raco test: non-empty stderr: #"--------------------\nERROR\nlocation:   db.rkt:1701:8\nname:       check-equal?\n\ntcp-connect: connection failed;\n host not found\n  address: www.dictionary.com\n  port number: 80\n  system error: Name or service not known; gai_err=-2\n--------------------\n--------------------\nERROR\nlocation:   db.rkt:1702:8\nname:       check-equal?\n\ntcp-connect: connection failed;\n host not found\n  address: www.dictionary.com\n  port number: 80\n  system error: Name or service not known; gai_err=-2\n--------------------\n--------------------\nERROR\nlocation:   db.rkt:1703:8\nname:       check-equal?\n\ntcp-connect: connection failed;\n host not found\n  address: www.dictionary.com\n  port number: 80\n  system error: Name or service not known; gai_err=-2\n--------------------\n--------------------\nERROR\nlocation:   db.rkt:1704:8\nname:       check-equal?\n\ntcp-connect: connection failed;\n host not found\n  address: www.dictionary.com\n  port number: 80\n  system error: Name or service not known; gai_err=-2\n--------------------\n--------------------\nERROR\nlocation:   db.rkt:1705:8\nname:       check-equal?\n\ntcp-connect: connection failed;\n host not found\n  address: www.dictionary.com\n  port number: 80\n  system error: Name or service not known; gai_err=-2\n--------------------\n--------------------\nERROR\nlocation:   db.rkt:1709:8\nname:       check-equal?\n\ntcp-connect: connection failed;\n host not found\n  address: www.dictionary.com\n  port number: 80\n  system error: Name or service not known; gai_err=-2\n--------------------\n"
```
<https://plt.eecs.northwestern.edu/release-pkg-build/server/built/test-fail/ipoe.txt>

</details>

**Comment by rfindler** [comment]
I think rackunit catches exceptions.

**Comment by samth** [comment]
In my testing, `amap` installs fine and passes all tests, so that installation failure seems not-problematic.

**Comment by samth** [comment]
`sdl2` failed on the most recent release pkg-build, but that was an actual now-fixed bug in the package.

**Comment by pmatos** [comment]
Is this complete? All the items in the initial post have now been ticked.

**Comment by sorawee** [comment]
IIUC, yes, and 7.8 is released already anyway. Closing.
